js_modules/dagster-ui/packages/ui-core/src/runs/AssetTagCollections.tsx

```diff
@@ -44,19 +52,26 @@ function useShowMoreDialog<T>(
   dialogTitle: string,
   items: T[] | null,
   renderItem: (item: T) => React.ReactNode,
+  sortFn?: (a: T, b: T) => number,
 ) {
   const [showMore, setShowMore] = React.useState(false);
 
+  let itemsToRender = items;
+
+  if (sortFn && items) {
+    itemsToRender = [...items].sort(sortFn);
+  }
+
```

Comment:
this will execute on every render, meaning a new array will be created an sorted on every render which is bad. You should memoize derived computations in React instead.